pkg/kubectl/cmd/config/get_context.go

```diff
@@ -0,0 +1,163 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+)
+
+type GetContextOptions struct {
+	configAccess clientcmd.ConfigAccess
+	nameOnly     bool
+	showHeaders  bool
+	contextNames []string
+	out          io.Writer
+}
+
+const (
+	get_context_long = `Displays one or many contexts from the kubeconfig file.`
+
+	get_context_example = `# List all the contexts in your kubeconfig file
+kubectl config get-context
+
+# Describe one context in your kubeconfig file.
+kubectl config get-context my-context`
+)
+
+// NewCmdGetContext creates a command object for the "get-context" action, which
+// retrieves one or more contexts from a kubeconfig.
+func NewCmdConfigGetContext(out io.Writer, configAccess clientcmd.ConfigAccess) *cobra.Command {
+	options := &GetContextOptions{configAccess: configAccess}
+
+	cmd := &cobra.Command{
+		Use:     "get-context [(-o|--output=)detail|name)]",
+		Short:   "Describe one or many contexts",
+		Long:    get_context_long,
+		Example: get_context_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			outputFormat := cmdutil.GetFlagString(cmd, "output")
+			outputTypes := map[string]bool{"": true, "json": false, "yaml": false, "wide": false, "name": true, "go-template": false, "go-template-file": false, "jsonpath": false, "jsonpath-file": false}
+			supported, ok := outputTypes[outputFormat]
+			if !ok {
+				cmdutil.CheckErr(fmt.Errorf("output must be one of '' or 'name': %q.", outputFormat))
+			}
+			if !supported {
+				fmt.Printf("--output %v is not available in kubectl config get-context; reseting to default output format\n")
+				cmd.Flags().Set("output", "")
+			}
+			cmdutil.CheckErr(options.Complete(cmd, args, out))
+			cmdutil.CheckErr(options.RunGetContext())
+		},
+	}
+	// TODO when the resource printer can print non-runtime objects use AddPrinterFlags here.
+	cmd.Flags().StringP("output", "o", "", "Output format. One of: json|yaml|wide|name|go-template=...|go-template-file=...|jsonpath=...|jsonpath-file=... See golang template [http://golang.org/pkg/text/template/#pkg-overview] and jsonpath template [http://releases.k8s.io/HEAD/docs/user-guide/jsonpath.md].")
```

Comment:
Suggest creating a function for `output` flag, and call that function here and in `AddPrinterFlags`